modules/explorer-core/src/main/scala/org/ergoplatform/explorer/db/repositories/DexOrdersRepo.scala

```diff
@@ -0,0 +1,157 @@
+package org.ergoplatform.explorer.db.repositories
+
+import cats.instances.try_._
+import fs2.Stream
+import org.ergoplatform.explorer.Err.RequestProcessingErr.{
+  DexBuyOrderAttributesFailed,
+  DexSellOrderAttributesFailed
+}
+import org.ergoplatform.explorer.db.algebra.LiftConnectionIO
+import org.ergoplatform.explorer.db.models.aggregates.{
+  DexBuyOrderOutput,
+  DexSellOrderOutput,
+  ExtendedOutput
+}
+import org.ergoplatform.explorer.{HexString, TokenId}
+import scorex.util.encode.Base16
+import sigmastate.Values.ByteArrayConstant
+import sigmastate.serialization.ErgoTreeSerializer
+import sigmastate.{SLong, Values}
+
+import scala.util.{Failure, Success, Try}
+
+/** [[ExtendedOutput]] for DEX sell/buy orders data access operations.
+  */
+trait DexOrdersRepo[D[_], S[_[_], _]] {
+
+  /** Get all unspent main-chain DEX sell orders
+    */
+  def getAllMainUnspentSellOrderByTokenId(
+    tokenId: TokenId
+  ): S[D, DexSellOrderOutput]
+
+  /** Get all unspent main-chain DEX buy orders
+    */
+  def getAllMainUnspentBuyOrderByTokenId(
+    tokenId: TokenId
+  ): S[D, DexBuyOrderOutput]
+}
+
+object DexOrdersRepo {
+
+  def apply[D[_]: LiftConnectionIO]: DexOrdersRepo[D, Stream] =
+    new Live[D]
+
+  final private class Live[D[_]: LiftConnectionIO] extends DexOrdersRepo[D, Stream] {
+
+    import org.ergoplatform.explorer.db.queries.{DexOrdersQuerySet => QS}
+
+    private val liftK = LiftConnectionIO[D].liftConnectionIOK
+
+    override def getAllMainUnspentSellOrderByTokenId(
+      tokenId: TokenId
+    ): Stream[D, DexSellOrderOutput] =
+      QS.getMainUnspentSellOrderByTokenId(
+          tokenId,
+          sellContractTemplate,
+          0,
+          Int.MaxValue
+        )
+        .stream
+        .map(eOut =>
+          DexSellOrderOutput(
+            eOut,
+            getTokenPriceFromSellOrderTree(eOut.output.ergoTree).get
+          )
+        )
+        .translate(liftK)
+
+    /** Get all unspent main-chain DEX buy orders
+      */
+    override def getAllMainUnspentBuyOrderByTokenId(
+      tokenId: TokenId
+    ): Stream[D, DexBuyOrderOutput] =
+      QS.getMainUnspentBuyOrderByTokenId(tokenId, buyContractTemplate, 0, Int.MaxValue)
+        .stream
+        .map(eOut =>
+          DexBuyOrderOutput(eOut, getTokenInfoFromBuyOrderTree(eOut.output.ergoTree).get)
```

Comment:
Using `.get` on Try is unacceptable, please handle business errors properly

Reply:
Done.